### ISSUE TITLE
Package Management features

### ISSUE THREAD

**Issue by NickSeagull** [neohaskell/NeoHaskell#32]
- install: Install packages
- uninstall: Uninstall packages
- update: Update packages
- list: List installed packages

**Comment by jinyus** [comment]
outdated: List outdated packages

**Comment by Daeraxa** [comment]
Is this specifically for development packages or global executable packages? You will have to forgive my inexperience with Haskell here in case many/none of the following are actually applicable.

If we are talking about system packages (also relevant in many cases for dev too) then it should have:
- `search/query` - to search for the package and return descriptions, versions etc. e.g. `cargo/npm/apt/dnf search`
- `help` - access to help file/documentation both for the package manager itself *and* for the package name given as an argument

If it is just for development then I assume it will need something to start a project. See: `cabal init`, `stack new`, `cargo new/init`, `npm init`.

And would also need the basic commands to add packages to the dependency file. See:
`cargo add`, `npm install` and the dev dependency equivalent flag e.g. `--save-dev`, `-dev`.

Then obviously commands to run, build, test etc. the package. I won't specify everything, you get the idea.

**Comment by NickSeagull** [comment]
Thanks @Daeraxa !

There are some other issues about the CLI defined as well here, in case you want to take a look https://github.com/neohaskell/NeoHaskell/milestone/2

**Comment by Daeraxa** [comment]
Right, I think I was hung up on the definition of "Package management" in a more general term, I've found lots of the things I mentioned in other sections now.
